src/index.js

```diff
@@ -15,7 +15,7 @@ function inlineKatex(options) {
     level: 'inline',
     start(src) { return src.indexOf('$'); },
     tokenizer(src, tokens) {
-      const match = src.match(/^\$+([^$\n]+?)\$+/);
+      const match = src.match(/^\$+((?:\n)?\S(?:.*\S)?(?:\n)?)\$+/);
```

Comment:
Actually there can be many opinions here. But if you see basic rules for usng markdown, like for bold you write `**bold**`, but if you  write `**bold **` or `** bold**` it wont work, only `**...**` works, now there are some functions which works with spaces also, like the inline code \`. but it has its own problem, for suppose, in the line below, just putting a backtick in wrong place will render different result :
abcd `code` ef`gh `code` ijkl

Here a backtick in the middle of efgh will render the output very differently, here i wanted to generate inline code only the `code` strings. Well we can escape the backtick with \\ , but currently we cant escape `$` with backtick like `\$`. We dont have that feature right now, not in the current extension version, not in my version, my version does not support \\$ fully, the current extension version supports \\$ outside of katex. but katex has its own syntax `\$` in order to render `$`, in my version, this support is partial, but `$\$$` does not work. This same code also wont work in current extension. But in my version, `$abcd\$efgh$` works fine, so basically its partial support, instead of just throwing error. later we may add a different functionality for `\$` so I think we should not allow space inside `$...$`. Well if we do allow, I doont know the `\$` inside katex does not work. Just like in the current version, `\$` is not working.

Also, syntax like `**...**` is more better than syntax like code (`), I'm using katex for a while, I think it as a better way, also some editors do the same. for example stackedit.io, which also doesn't support `\$` inside katex. But stricting `$...$` and not allowing space is not a problem I think, it just makes it better to detect `$...$`. But I think both way is enough to work with, now you decide and tell me what  should we do, then I will update the code 💯

Reply:
Markdown is meant to be easy to read for bold it makes sense that `this is **bold**` is easier to read than `this is ** bold **` with bold it is easier to understand what is meant when there are no spaces. For katex I think `$ c = \\pm\\sqrt{a^2 + b^2} $` is easier to read than `$c = \\pm\\sqrt{a^2 + b^2}$` that `$c` can be easily confused as one variable at a glance.

I think to prevent something like `$300 $400` being seen as katex we should require a space outside the `$...$` so `This is katex:$ c = \\pm\\sqrt{a^2 + b^2} $` does not count as katex there will have to be a space between `: $` so it is easier to read.

Reply:
also anything that parses text with newlines should be in a block level tokenizer. We don't want to allow newlines with other content like 

```
this is katex: $
c = \\pm\\sqrt{a^2 + b^2}
$
```

We should only allow newlines when the first `$` is at the beginning of the line like

```
this is katex:
$
c = \\pm\\sqrt{a^2 + b^2}
$
```

Reply:
> Markdown is meant to be easy to read for bold it makes sense that `this is **bold**` is easier to read than `this is ** bold **` with bold it is easier to understand what is meant when there are no spaces. For katex I think `$ c = \\pm\\sqrt{a^2 + b^2} $` is easier to read than `$c = \\pm\\sqrt{a^2 + b^2}$` that `$c` can be easily confused as one variable at a glance.
> 
> I think to prevent something like `$300 $400` being seen as katex we should require a space outside the `$...$` so `This is katex:$ c = \\pm\\sqrt{a^2 + b^2} $` does not count as katex there will have to be a space between `: $` so it is easier to read.

Well we can use `\$300 \$400` to prevent silly issue like that, also we can allow space, no problem here, the current extension is actually ok to work with, I agree, but some features are not being supported, like `$\fcolorbox{red}{aqua}{$\frac{b}{2}\sqrt{a^2-\frac{b^2}{4}}$}$` or this :
```tex
$$
\fcolorbox{red}{aqua}{$\frac{b}{2}\sqrt{a^2-\frac{b^2}{4}}$}
$$
```
I think my version of the block level is ok, its supporting the codes, but the current inline version does not support the same code inline because of `$...$....$...$` and in my vversion, this is matched, so in my version its a bug, the regexp cant detect 2 different katex, rather matches as a same katex, so `katex : $\pi$ and $\theta$` will not work in my version, it will match `\pi$ and $\theta`. in many cases there are multiple inline katex in single line, but block katex has to be in different line, so I guess we need to get extra thing here, like we can introduce another matching regex, for something like \`$...$\`, in order to allow a katex inside another katex, and if the case is not, then normal katex like `$...$`, but the user will have both functionality to work with. How abouth this idea? And also in this version of extension the `\$` does not work inside katex, how do we add this support? `\$` is valid katex code.